src/finch/interface/lazy.py

```diff
@@ -959,6 +1002,419 @@ def vecdot(x1, x2, /, *, axis=-1) -> LazyTensor:
+        axis: Axis along which to concatenate (default=0)
+
+    Returns:
+        Concatenated tensor as LazyTensor
+    """
+    arrays = [defer(arr) for arr in arrays]
+
+    if axis is None:
+        # flatten all tensors
+        for i, t in enumerate(arrays):
+            arrays[i] = flatten(t)
+        axis = 0
+
+    # Convert axis to positive index and validate
+    ndim = arrays[0].ndim
+    axis = normalize_axis_index(axis, ndim)
+    computed_arrays = tuple(_compute(arr) for arr in arrays)
+    concat_tensor = ConcatTensor(*computed_arrays, axis=axis)
+    # Create a LazyTensor that represents the concatenation
+    return elementwise(identity, defer(concat_tensor))
+
+
+class SplitDimsTensor(Tensor):
+    """
+    Tensor representing a dimension split operation.
+    """
+
+    def __init__(self, tensor, axis: int, shape: tuple[int, ...]):
+        """
+        Args:
+            tensor: The input tensor to split
+            axis: The axis to split
+            shape: The new shape for the split dimensions
+        """
+        self.tensor = tensor
+        self.axis = normalize_axis_index(axis, tensor.ndim)
+
+        # Validate that the product of new dimensions equals the original dimension
+        shape_product = np.prod(shape)
+        if shape_product != tensor.shape[self.axis]:
+            raise ValueError(
+                f"Cannot split dimension of size {tensor.shape[self.axis]} "
+                f"into shape {shape}. Product of new dimensions "
+                f"({shape_product}) must equal original size."
+            )
+
+        # Create new shape by replacing the axis dimension with the split dimensions
+        self._shape = tensor.shape[: self.axis] + shape + tensor.shape[self.axis + 1 :]
+        self.split_shape = shape
+        self._ndim = len(self._shape)
+        self._element_type = element_type(tensor)
+        self.dtype = self._element_type
+
+    def __getitem__(self, idxs: tuple[int, ...]):
+        """
+        Args:
+            idxs: Indices to access the split tensor
+        Returns the element at the specified indices by mapping back to original tensor
+        """
+        if not isinstance(idxs, tuple):
+            idxs = (idxs,)
+
+        # Extract the indices for the split dimensions
+        split_start = self.axis
+        split_end = self.axis + len(self.split_shape)
+        split_idxs = idxs[split_start:split_end]
+
+        # Convert multi-dimensional split indices back to linear index
+        linear_idx = 0
+        multiplier = 1
+        for i in reversed(range(len(self.split_shape))):
+            linear_idx += split_idxs[i] * multiplier
+            multiplier *= self.split_shape[i]
+
+        # Reconstruct the original indices
+        original_idxs = idxs[: self.axis] + (linear_idx,) + idxs[split_end:]
+
+        return self.tensor[original_idxs]
+
+    @property
+    def format(self):
+        return LazyTensorFormat(
```

Comment:
similarly for this tensor, make a `SplitTensorFormat`. It would be great if all of these new formats inherited from a shared `WrapperTensorFormat`.